frontend/vite.config.ts

```diff
@@ -1,8 +1,30 @@
-import { defineConfig } from 'vite'
+import { defineConfig, PluginOption, splitVendorChunkPlugin } from 'vite'
 import svgr from 'vite-plugin-svgr'
 import react from '@vitejs/plugin-react-swc'
+import { visualizer } from 'rollup-plugin-visualizer'
 
 // https://vitejs.dev/config/
 export default defineConfig({
-  plugins: [svgr(), react()],
+  plugins: [svgr(), react(), splitVendorChunkPlugin(), visualizer() as PluginOption],
+  build: {
+    sourcemap: true,
+    rollupOptions: {
+      output: {
+        manualChunks(id: string) {
+          if (id.includes('ethers')) {
+            return 'ethers'
+          }
+          if (id.includes('recharts')) {
+            return 'recharts'
+          }
+          if (id.includes('lodash')) {
+            return 'lodash'
+          }
+          if (id.includes('react-dom')) {
+            return 'react-dom'
+          }
+        },
```

Comment:
What is the benefit of this? It doesn't look like it will load these modules on demand

Reply:
That was not my intention. Vite build process complains(just warning though, see first output in PR desc) if a single bundled .js file, is over 500kb uncompressed. It is just a way to specify splitting https://rollupjs.org/configuration-options/#output-manualchunks

But I can look into to dynamically load `recharts`, as it is only used in results page.

Reply:
Did create dynamic import for `recharts`(https://github.com/oasisprotocol/dapp-votee/tree/ml/split-chunks-dynamic-recharts), but then tree shaking doesn't apply:
![Screenshot_28-3-2024_61135_](https://github.com/oasisprotocol/dapp-votee/assets/9722540/8eb4b26b-9265-4352-89c0-83d55fabc758)

Also did try to dynamically import pages (https://github.com/oasisprotocol/dapp-votee/tree/ml/split-chunks-dynamic-pages) but that creates too much chunks IMO:
![Screenshot_28-3-2024_6152_](https://github.com/oasisprotocol/dapp-votee/assets/9722540/1a444dfb-1ff7-47c0-916e-413c8acb30de)

I would just go with the current solution.

Reply:
So far I'm convinced that the current solution doesn't do anything beneficial:
- loads and parses the same amount of javascript
- it adds a little overhead with multiple connections
- it adds a little complexity to the configuration
- it prevents vite from analyzing and showing a useful-but-ignorable warning to consider dynamic loading

Am I missing something and are you seeing faster load time?

Reply:
I can try tomorrow and will post the results, but essentially, I am guessing that load time should be a bit faster(still very small app) - due to parallel fetching of the resources.

Reply:
Comparison:

`master`:
![image](https://github.com/oasisprotocol/dapp-votee/assets/9722540/0f0904fd-36ac-4ce5-b2ab-ec0bfa686620)

`ml/split-chunks`:
![image](https://github.com/oasisprotocol/dapp-votee/assets/9722540/777e3c8e-90e4-40af-a82c-9b843824f673)

There is like +-100ms difference. Which could as well be within margin of an "error". I would need to perform tests multiple times to see the actual difference.

Reply:
On deployed site I measure:
- https://dapp-votee-try-vite-default-6054235.pages.dev/
  unthrottled network tab Finish: 480ms
  Lighthouse Largest Contentful Paint 1.6s - 2.4s
- https://dapp-votee-try-vite-manual-split-6054235.pages.dev/
  unthrottled network tab Finish: 477ms
  Lighthouse Largest Contentful Paint 1.4s - 2.6s :shrug:

Reply:
![image](https://github.com/oasisprotocol/dapp-votee/assets/9722540/b4db38ad-5f0e-4bc1-9944-6a52ae2a2a90)

To me it's quite similar... So merge as is? Or do I revert to single chunk for .js? Because it looks like, it doesn't matter anyway.

Reply:
I would still revert: "chunk should not be larger than 500kB" in commit message is misleading

Reply:
Done